air/src/range.rs

```diff
@@ -0,0 +1,68 @@
+use vm_core::RANGE_CHECK_TRACE_OFFSET;
+
+use super::{
+    utils::is_binary, Assertion, EvaluationFrame, Felt, FieldElement, TransitionConstraintDegree,
+};
+
+// CONSTANTS
+// ================================================================================================
+
+pub const COL_IDX_T: usize = RANGE_CHECK_TRACE_OFFSET;
+pub const COL_IDX_S0: usize = RANGE_CHECK_TRACE_OFFSET + 1;
+pub const COL_IDX_S1: usize = RANGE_CHECK_TRACE_OFFSET + 2;
+pub const COL_IDX_V: usize = RANGE_CHECK_TRACE_OFFSET + 3;
+
+// CONSTRAINT DEGREES
+// ================================================================================================
+
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    vec![
+        // Selector flags must be binary: t, s0, s1
+        TransitionConstraintDegree::new(2),
+        TransitionConstraintDegree::new(2),
+        TransitionConstraintDegree::new(2),
+        // Transition from 8-bit to 16-bit section of range check table occurs at most once.
+        TransitionConstraintDegree::new(2),
+        // Enforce values of column v before and after 8-bit to 16-bit transition.
+        TransitionConstraintDegree::new(3),
+        TransitionConstraintDegree::new(3),
+    ]
+}
+
+// BOUNDARY CONSTRAINTS
+// ================================================================================================
+
+pub fn get_assertions_first_step(result: &mut Vec<Assertion<Felt>>) {
+    result.push(Assertion::single(COL_IDX_V, 0, Felt::ZERO));
+}
+
+pub fn get_assertions_last_step(result: &mut Vec<Assertion<Felt>>, step: usize) {
+    result.push(Assertion::single(COL_IDX_V, step, Felt::new(65535)));
+}
+
+// TRANSITION CONSTRAINTS
+// ================================================================================================
+
+pub fn enforce_constraints<E: FieldElement<BaseField = Felt>>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+) {
+    let current = frame.current();
+    let next = frame.next();
+
+    // --- Constrain the selector flags. ----------------------------------------------------------
+    result[0] = is_binary(current[COL_IDX_T]);
+    result[1] = is_binary(current[COL_IDX_S0]);
+    result[2] = is_binary(current[COL_IDX_S1]);
+
+    // --- Constrain the transition from 8-bit to 16-bit section of the table. --------------------
+    // Values in column t can "flip" from 0 to 1 only once.
+    result[3] = current[COL_IDX_T] * (E::ONE - next[COL_IDX_T]);
+
+    let flip = (E::ONE - current[COL_IDX_T]) * next[COL_IDX_T];
+    // When column t "flips", column v must equal 255.
+    result[4] = flip * (current[COL_IDX_V] - E::from(255u64));
+
+    // When column t "flips", the next value column v must be reset to 0.
+    result[5] = flip * next[COL_IDX_V];
```

Comment:
A few small comments:

* nit: let's use `255_u8` instead of `255u64`. This may be a tiny bit more efficient, and I think the format with using underscores is more readable (I'm going to try to transition to it myself).
* I think my note has an error. Specifically, in the final set of constraints I'm missing `(v' - v) * (v' - v - 1)` constraint for the 8-bit portion of the table. But could you verify that it is actually needed?

Reply:
Yep, I think we definitely need this & I've added it. (The u8 change is done as well). 

However, now that we're using a flag, the constraints here are a bit busier. Do you want me to take a step back and start thinking about how we want to handle flags at this point? Or should it we leave it to get updated as we address the more general cases during future constraint handling?

Reply:
Great! I think we can leave it as if for now and revisit later.